kerfuffle/jobs.h

```diff
@@ -224,7 +224,7 @@ class KERFUFFLE_EXPORT AddJob : public Job
     Q_OBJECT
 
 public:
-    AddJob(QList<Archive::Entry*> &files, const CompressionOptions& options, ReadWriteArchiveInterface *interface);
+    AddJob(const QList<Archive::Entry*> &files, const Archive::Entry *destination, const CompressionOptions& options, ReadWriteArchiveInterface *interface);
```

Comment:
If you se a default `destination` here, you don't need to pass `new Archive::Entry(this)` all over the place (when you don't want to add files to a specific destination).